lib/pengine/utils.c

```diff
@@ -447,6 +447,79 @@ find_existing_action(const char *key, pe_resource_t *rsc, pe_node_t *node,
     return action;
 }
 
+/*!
+ * \internal
+ * \brief Create a new action object
+ *
+ * \param[in] key        Action key
+ * \param[in] task       Action name
+ * \param[in] rsc        Resource that action is for (if any)
+ * \param[in] node       Node that action is on (if any)
+ * \param[in] optional   Whether action should be considered optional
+ * \param[in] for_graph  Whether action should be recorded in transition graph
+ * \param[in] data_set   Cluster working set
+ *
+ * \return Newly allocated action
+ * \note This function takes ownership of \p key. It is the caller's
+ *       responsibility to free the return value with pe_free_action().
+ */
+static pe_action_t *
+new_action(char *key, const char *task, pe_resource_t *rsc, pe_node_t *node,
+           bool optional, bool for_graph, pe_working_set_t *data_set)
+{
+    pe_action_t *action = calloc(1, sizeof(pe_action_t));
+
+    CRM_ASSERT(action != NULL);
+
+    action->rsc = rsc;
+    action->task = strdup(task); CRM_ASSERT(action->task != NULL);
+    action->uuid = key;
```

Comment:
Did you mean to strip the strdup off this line?

Reply:
Yes, that's what the commit message is referring to ... previously custom_action() would strdup() the provided key if this block were hit, and always free the provided key. Now, we use the provided key here directly and only free it if this isn't hit.